Server/Server.Core/Server.Core.Static/SettingsManager.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Settings.hpp"
+
+namespace Server::Builder
+{
+class SettingsManager
+{
+public:
+    SettingsManager() = default;
+
+    explicit SettingsManager(std::unique_ptr<Settings> settings) : _settings(std::move(settings)) {}
```

Comment:
Reason

Passing a smart pointer transfers or shares ownership and should only be used when ownership semantics are intended. A function that does not manipulate lifetime should take raw pointers or references instead.

Passing by smart pointer restricts the use of a function to callers that use smart pointers. A function that needs a widget should be able to accept any widget object, not just ones whose lifetimes are managed by a particular kind of smart pointer.

Passing a shared smart pointer (e.g., std::shared_ptr) implies a run-time cost.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r30-take-smart-pointers-as-parameters-only-to-explicitly-express-lifetime-semantics

Reply:
Hm... I'll think about it and try to use it.